### PR TITLE
#534 added deprecation warning

### LINKED ISSUE #534: Display a deprecation warning
The version and approval features in Kio are deprecated and will be removed soon. Users of Yourturn should see a warning about this deprecation.

- [ ] display a warning on the application version pages
- [ ] display a warning on the approval pages
- [ ] maybe display a dismissible warning already on the landing page

TODO: the exact wording of the deprecation warning

### PR DESCRIPTION
**Pull request by neremic**
fixes #534

### CODE REVIEW

File: client/lib/application/src/approval-form/approval-form.jsx [hunk 1]
```diff
@@ -133,6 +133,11 @@ class ApprovalForm extends React.Component {
         const {applicationId, versionId, application, approvalTypes, approvals, editable, userInfos} = this.props,
         LINK_PARAMS = {applicationId, versionId};
         return <div className='approvalForm'>
+                    <div style={{backgroundColor: '#f08532'}}>
```
Comment (gagoman) [inline client/lib/application/src/approval-form/approval-form.jsx]: The same component is copied 3 times. Can we consider separate component creation?

File: client/lib/yourturn/src/app.jsx [hunk 2]
```diff
@@ -2,21 +2,43 @@ import React from 'react';
...
                         <Sidebar
-                            activeRoute={props.location.pathname} />
+                            activeRoute={this.props.location.pathname} />
                     </div>
                     <div className='grid-col'>
+                        {this.state.show ?
+                            <div style={{backgroundColor: '#f08532'}} onClick={this.dismiss}>
+                                <h4 style={{color: '#FFF', margin: '20px'}}>
+                                    Please note that the GitHub approval flow obsoletes Kio versions, i.e. you don't have to maintain version information in Kio anymore. The corresponding API endpoints will be disabled on July 31st, 2017.<br />
+                                    Click to dismiss!
```
Comment (harti2006) [inline client/lib/yourturn/src/app.jsx]: From UX perspective, a "X" in the top right corner would be great to dismiss the alert

File: client/lib/application/src/approval-form/approval-form.jsx (hunk 1 above)
Comment (neremic) [inline client/lib/application/src/approval-form/approval-form.jsx]: Would make sense but this is only there for the next few weeks. So I consider this to be ok.

File: client/lib/yourturn/src/app.jsx (hunk 2 above)
Comment (neremic) [inline client/lib/yourturn/src/app.jsx]: Well, ok... will do.

**Comment by gagoman** [comment]
@neremic please fix lint issues. They are preventing the merge.

**Comment by neremic** [comment]
:+1:

**Comment by harti2006** [comment]
I'd like discuss the current implementation in the team.

**Comment by hjacobs** [comment]
@harti2006 ping :smile:

**Comment by harti2006** [comment]
@hjacobs there is a small fix pending. Then we will most probably release this change this week

File: client/lib/yourturn/src/app.jsx [hunk 3]
```diff
@@ -1,23 +1,41 @@
...
-                <div className='grid with-gutter'>
+        <div className='grid with-gutter'>
                     <div className='grid-col col-1-4'>
                         <Sidebar
-                            activeRoute={props.location.pathname} />
+                            activeRoute={this.props.location.pathname} />
                     </div>
                     <div className='grid-col'>
+                        {this.props.location.pathname === '/' ?
+                            <DeprecationWarning dismissable={false} /> : null
```
Comment (harti2006) [inline client/lib/yourturn/src/app.jsx]: Can you please remove all the code, that handles the dismissing stuff? Sorry for the extra work, but it doesn't fell right to merge unused code.

**Comment by neremic** [comment]
:+1:

**Comment by harti2006** [comment]
:+1:

**Pull request merged**